core/src/main/java/org/elasticsearch/index/mapper/internal/TypeFieldMapper.java

```diff
@@ -105,6 +126,29 @@ public String typeName() {
             return CONTENT_TYPE;
         }
 
+        public boolean fielddata() {
+            return fielddata;
+        }
+
+        public void setFielddata(boolean fielddata) {
+            checkIfFrozen();
+            this.fielddata = fielddata;
+        }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder() {
+            if (hasDocValues()) {
+                return new DocValuesIndexFieldData.Builder();
+            }
+            assert indexOptions() != IndexOptions.NONE;
+            if (fielddata) {
+                return new PagedBytesIndexFieldData.Builder(StringFieldMapper.Defaults.FIELDDATA_MIN_FREQUENCY,
+                    StringFieldMapper.Defaults.FIELDDATA_MAX_FREQUENCY,
+                    StringFieldMapper.Defaults.FIELDDATA_MIN_SEGMENT_SIZE);
```

Comment:
can you s/StringFieldMapper/TextFieldMapper/ since we aim at removing StringFieldMapper in the future?